lib/controllers/messages.ts

```diff
@@ -569,6 +558,9 @@ export function GetMessageHandler(
 
     const retrievedMessage = maybeDocument.value;
 
+    // whether the user is a trusted application (i.e. can access all messages for any recipient)
+    const canListMessages = userAuth.groups.has(UserGroup.ApiMessageList);
```

Comment:
I don't get it. This is not a check by itself but just an assignment of a variable (the process will arrive at this point anyway, even if we move it at the beginning)

Reply:
yes I mean move back the check at the beginning - you have merged the two checks (one before the query and one after the query) to happen both after the query, the check on `canListMessages === true` should happen before the query gets issued

Reply:
I still don't get it. I haven't merged any check, I've just deleted an unused one (on userAttributes.service). What should happen in case canListMessages === true (or false) before the query ?

Reply:
sorry I new see what you mean, I thought this was `GetMessages`, I think we're missing the check that the user has the `ApiMessageRead` OR the `ApiMessageList` group before the query

the logic should be:

1. can read it's own messages (`ApiMessageRead`) or all messages (`ApiMessageList`) ? -> reject if not
2. read message
3. can read all messages OR he's the sender of the message? -> reject if not  
4. return the message

Reply:
The check is already there, in the GetMessage wrapper method:

```
    AzureApiAuthMiddleware(
      new Set([UserGroup.ApiMessageRead, UserGroup.ApiMessageList])
    ),
```